File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.java
class Solution {
    public String solution(String s) {
        String answer = "";
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if (c == ' ' || Character.isDigit(c)) {
                answer += c;
                continue;
            }
            if (i == 0 || (i > 0 && s.charAt(i - 1) == ' ')) {
                answer += Character.toUpperCase(c);
            } else {
                answer += Character.toLowerCase(c);
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.py
def solution(s):
    answer = ''
    for i in range(len(s)):
        if s[i] == ' ' or s[i].isdigit():
            answer += s[i]
            continue
        if i == 0 or (i > 0 and s[i - 1] == ' '):
            answer += s[i].upper()
        else:
            answer += s[i].lower()
    return answer